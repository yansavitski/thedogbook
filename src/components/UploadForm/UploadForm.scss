@import 'src/global';

.UploadForm {
  box-sizing: border-box;
  cursor: pointer;
  display: flex;

  &-input {
    display: none;
  }

  &-icon {
    margin-right: $baseOffset/2;
  }
}
