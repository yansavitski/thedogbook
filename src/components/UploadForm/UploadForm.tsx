import './UploadForm.scss'

import classNames from 'classnames'
import { Button } from 'components/Button/Button'
import React, { ChangeEvent, useCallback, useRef as useReference } from 'react'
import { FiUpload as UploadIcon } from 'react-icons/fi'

interface UploadFormProps {
  className?: string
  disabled?: boolean
  onLoadFile: (event: ChangeEvent<HTMLInputElement>) => void
}

export const UploadForm: React.FunctionComponent<UploadFormProps> = React.memo(
  ({ className, onLoadFile, disabled = false }: UploadFormProps) => {
    const fileInputReference = useReference<HTMLInputElement>(null)
    const handleUploadClick = useCallback(() => {
      fileInputReference.current?.click()
    }, [fileInputReference])

    return (
      <form className={classNames('UploadForm', className)}>
        <input
          className="UploadForm-input"
          type="file"
          name="fileToUpload"
          accept="image/*"
          ref={fileInputReference}
          onChange={onLoadFile}
          disabled={disabled}
        />

        <Button
          className="UploadForm-button"
          onClick={handleUploadClick}
          disabled={disabled}
        >
          <>
            <UploadIcon
              size={24}
              className="UploadForm-icon"
              title="Upload image"
            />
            Upload Image
          </>
        </Button>
      </form>
    )
  },
)

UploadForm.displayName = 'UploadForm'
