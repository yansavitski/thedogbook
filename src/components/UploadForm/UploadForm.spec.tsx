import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import React from 'react'

import { UploadForm } from './UploadForm'

describe('<UploadForm />', () => {
  it('renders correctly', () => {
    expect.assertions(1)
    const tree = shallow(<UploadForm onLoadFile={(): void => {}} />)

    expect(toJson(tree)).toMatchSnapshot()
  })

  it('renders correctly with optional props', () => {
    expect.assertions(1)
    const tree = shallow(
      <UploadForm className="preview" onLoadFile={(): void => {}} disabled />,
    )

    expect(toJson(tree)).toMatchSnapshot()
  })
})
