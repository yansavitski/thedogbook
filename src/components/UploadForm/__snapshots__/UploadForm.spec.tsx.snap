// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UploadForm /> renders correctly 1`] = `
<form
  className="UploadForm"
>
  <input
    accept="image/*"
    className="UploadForm-input"
    disabled={false}
    name="fileToUpload"
    onChange={[Function]}
    type="file"
  />
  <Component
    className="UploadForm-button"
    disabled={false}
    onClick={[Function]}
  >
    <FiUpload
      className="UploadForm-icon"
      size={24}
      title="Upload image"
    />
    Upload Image
  </Component>
</form>
`;

exports[`<UploadForm /> renders correctly with optional props 1`] = `
<form
  className="UploadForm preview"
>
  <input
    accept="image/*"
    className="UploadForm-input"
    disabled={true}
    name="fileToUpload"
    onChange={[Function]}
    type="file"
  />
  <Component
    className="UploadForm-button"
    disabled={true}
    onClick={[Function]}
  >
    <FiUpload
      className="UploadForm-icon"
      size={24}
      title="Upload image"
    />
    Upload Image
  </Component>
</form>
`;
